issue: 1066306 Add VMA Poll IOMUX mode support

igor-ivanov opened the pull request.

igor-ivanov committed:
issue: 1066306 Code cleanup

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>

igor-ivanov committed:
issue: 1066306 Add branch prediction macro

These in turn instruct the compiler which branch of the program
should be faster and which may execute slower.

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>

igor-ivanov committed:
issue: 1066306 Add VMA Poll IOMUX mode support

This reverts commit 84824a167ee07bf4f18ec26c83939b31f8c0b714.

igor-ivanov committed:
issue: 1066306 Replace vmapoll with socketxtreme

Some VMA extra API interfaces are renamed. This change adapts
sockperf application to this.
Replace '--vmapoll|-v' iomux usage with '--socketxtreme|-x'

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>

liranoz12 commented:
@AlaAyesh Please review

rafiw commented on configure.ac:
```diff
@@ -75,12 +75,7 @@ SP_CHECK_CXXFLAGS_APPEND([OUR_CXXFLAGS], [\
+have_vma_api=yes
 AS_IF([test "x${have_vma_api}" == "xyes"],
     [AC_CHECK_HEADER([mellanox/vma_extra.h],
         [AC_DEFINE([USING_VMA_EXTRA_API],[1],[[Enable using VMA extra API]])],
```
@igor-ivanov please also remove
AC_MSG_CHECKING(
    [for vma extra api])
AC_MSG_RESULT([${have_vma_api}])
and the AS_IF
so we will have only 
AC_CHECK_HEADER([mellanox/vma_extra.h],
    [AC_DEFINE([USING_VMA_EXTRA_API],[1],[[Enable using VMA extra API]])],
    [AC_MSG_NOTICE([vma_extra.h file not found])]
    [have_vma_api=no])

igor-ivanov replied:
I would like to keep one as is because from my point it provides more friendly output for end user
As for AS_IF I saved this to have quickly configure sockperf w/o VMA API for development usage just assigning `have_vma_api=no`

rafiw replied:
ok

rafiw replied:
my opinion change to default true

AlaAyesh commented:
I have reviewed the PR and it is ok.

igor-ivanov committed:
issue: 1066306 Add silent VMA API configuration

Now sockperf is compiled with VMA EXTRA API by defaulr if VMA is
detected on the system.

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>

igor-ivanov committed:
issue: 1066306 Exit on VMA API initialization failure

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>
